### ISSUE TITLE
Ox injects strings into Element nodes array

### ISSUE THREAD

**Issue by quicksurf** [ohler55/ox#151]
I have this sample xml (this is a small fragment of a much larger xml document, but reproduces the issue):

```
<?xml version="1.0" encoding="UTF-8"?>
<perf_report_response status="quarantined">
  <tasks>
    <task id="1711061" type="COMM_CHECK" created_at="1473291480" priority="5" params="">
    </task>
  </tasks>
</perf_report_response>
```

and when I do an Ox.parse on it, I get this:

```
#<Ox::Document:0x007fa4a4117150 @attributes={:version=>"1.0", :encoding=>"UTF-8"}, @nodes=[#<Ox::Element:0x007fa4a4117010 @value="perf_report_response", @attributes={:status=>"quarantined"}, @nodes=[#<Ox::Element:0x007fa4a4116f48 @value="tasks", @nodes=[#<Ox::Element:0x007fa4a4116ea8 @value="task", @attributes={:id=>"1711061", :type=>"COMM_CHECK", :created_at=>"1473291480", :priority=>"5", :params=>""}, @nodes=["\n    "]>]>]>]>
```

The issue I'm having is that the code that steps through this is expecting the `@nodes` array to either be empty, or contain Ox node data types, not 1 or more strings mixed in there as you can see by the last nodes array at the end of the line. On the much larger xml document, these strings are all over the place in the various `@nodes` arrays.

This xml is xml that I'm receiving from a source that I have no control over.  Right now, I'm working around it by doing a "is_a?" type check for the string data type when iterating over the nodes array, but would very much rather not have to.  It strikes me as a bug that those strings would be in the `@node` arrays.

**Comment by ohler55** [comment]
Try Ox.load with an option of :skip => :skip_white.

**Comment by quicksurf** [comment]
OK.  That works.  Thank you.

I guess this is a little off topic, but why have multiple ways to load/parse a document?  Is one newer and the other older and bound to be phased out at some point?

**Comment by ohler55** [comment]
Having two ways to parse or load follows the JSON gem convention. It would not have been my first choice in designing the API.

**Comment by quicksurf** [comment]
Ugh... It's not your fault, but stuff like that really makes me have a love/hate relationship with Ruby.  Clean and simple is better, especially when it comes to APIs and usage.  Multiple ways to do the same/similar thing just makes for more confusion.  I guess I just don't think or code the same way most other Ruby developer do.

At any rate, thanks.  That change straightened things out and was easy enough to do.  Things are humming along quite nicely and I was able to take out the conditional checking code.

On a different note, we're looking to move a bunch of our code over from yajl to oj, in the next 3-4 months (likely), so you'll probably see me pop up over there with questions in the not too distant future.  Right now we're doing require 'yajl/json_gem' and basically acting like we did a require 'json'.  I was hoping there'd be something simple like that with oj, but it looks like there's a different way to do that with oj?  Need to look closer at it.
